src/release.js

```diff
@@ -58,9 +60,32 @@ module.exports = function taoExtensionReleaseFactory(params = {}) {
     let gitClient;
     let githubClient;
     let taoInstance;
+    let npmPackage;
 
     return {
 
+        /**
+         * Run the build command of the package so we release latest build
+         */
+        async buildPackage() {
```

Comment:
each package is supposed to build itself when running publish, this is the role of the pacakge's `prepare` action, like  https://github.com/oat-sa/tao-item-runner-fe/blob/master/package.json#L20
the `prepare` script is called after install and before publish (see https://docs.npmjs.com/misc/scripts#description) it's an official step of the npm lifecycle, when `build` is an internal convention

Reply:
Ok, understood, but this function is run as part of the github release phase. It's the equivalent of the `grunt bundle` command which is run on traditional extensions. It will guarantee that the package-lock.json is updated to match the package.json, catch unbuildable code, and generate any artifacts which need to be committed to the repo at release time.

Reply:
For an npm package this is not necessary otherwise it means the package cannot be published in its current state. This is due to the difference between the github repo (the source) and the npm package (built runtime/artifact) and the transition between the source state to the artifact state is `npm install` and `npm publish`